cfme/middleware/provider.py

```diff
@@ -87,3 +89,14 @@ def num_server(self):
     @num_server.variant('ui')
     def num_server_ui(self):
         return int(self.get_detail("Relationships", "Middleware Servers"))
+
+    def nav_to_provider_view(self):
+        self.load_details(self)
+
+    def nav_to_provider_detailed_view(self):
+        if not self._on_detail_page():
+            sel.force_navigate('middleware_provider_detail', context={
+                'provider': self})
+
+    def _on_detail_page(self):
+        return sel.is_displayed_text("{} (Summary)".format(self.name))
```

Comment:
This should call `ensure_browser_open()` before is_displayed otherwise it will raise an exception if the browser was not opened yet before.